samples/snippets/src/main/java/com/example/bigtable/AuthorizedViewExample.java

```diff
@@ -0,0 +1,323 @@
+        for (ByteString qualifierPrefix : entry.getValue().getQualifierPrefixes()) {
+          System.out.printf("Column Family: %s, Qualifier Prefix: %s%n", entry.getKey(),
+              qualifierPrefix.toStringUtf8());
+        }
+        for (ByteString qualifier : entry.getValue().getQualifiers()) {
+          System.out.printf("Column Family: %s, Qualifier: %s%n", entry.getKey(),
+              qualifier.toStringUtf8());
+        }
+      }
+    } catch (NotFoundException e) {
+      System.err.println(
+          "Failed to retrieve metadata from a non-existent authorized view: " + e.getMessage());
+    }
+    // [END bigtable_get_authorized_view]
+    return authorizedView;
+  }
+
+  /**
+   * Demonstrates how to list all authorized views within a table.
+   */
+  public List<String> listAllAuthorizedViews() {
+    System.out.printf("%nListing authorized views in table %s%n", tableId);
+    // [START bigtable_list_authorized_views]
+    List<String> authorizedViewIds = new ArrayList<>();
+    try {
+      authorizedViewIds = adminClient.listAuthorizedViews(tableId);
+      for (String authorizedViewId : authorizedViewIds) {
+        System.out.println(authorizedViewId);
+      }
+    } catch (NotFoundException e) {
+      System.err.println(
+          "Failed to list authorized views from a non-existent table: " + e.getMessage());
+    }
+    // [END bigtable_list_authorized_views]
+    return authorizedViewIds;
+  }
+
+  /**
+   * Demonstrates how to delete an authorized view.
+   */
+  public void deleteAuthorizedView() {
+    System.out.printf("%nDeleting authorized view %s in table %s%n", authorizedViewId, tableId);
+    // [START bigtable_delete_authorized_view]
+    try {
+      adminClient.deleteAuthorizedView(tableId, authorizedViewId);
+      System.out.printf("AuthorizedView: %s deleted successfully%n", authorizedViewId);
+    } catch (NotFoundException e) {
+      System.err.println("Failed to delete a non-existent authorized view: " + e.getMessage());
+    }
+    // [END bigtable_delete_authorized_view]
+  }
+
+  /**
+   * Demonstrates how to write some rows to an authorized view.
+   */
+  public void writeToAuthorizedView() {
+    // [START bigtable_authorized_view_write_rows]
+    try {
+      System.out.println("\nWriting to authorized view");
+      String[] names = {"World", "Bigtable", "Java"};
+      for (int i = 0; i < names.length; i++) {
+        String greeting = "Hello " + names[i] + "!";
+        RowMutation rowMutation =
+            RowMutation.create(AuthorizedViewId.of(tableId, authorizedViewId), ROW_KEY_PREFIX + i)
+                .setCell(COLUMN_FAMILY, COLUMN_QUALIFIER_NAME, names[i])
+                .setCell(COLUMN_FAMILY, COLUMN_QUALIFIER_GREETING, greeting);
+        dataClient.mutateRow(rowMutation);
+        System.out.println(greeting);
+      }
+    } catch (NotFoundException e) {
+      System.err.println("Failed to write to non-existent authorized view: " + e.getMessage());
```

Comment:
Maybe update this to "Failed to write to non-existent table or authorized view: " .. cause I think if the table doesn't exist, it would also be NotFound?

Reply:
Hmmm but here we are targeting the authorized view only. Therefore, in the AFE we would perform spanner lookups for the authorized view. So if the parent table doesn't exist, the NotFound error returned from the AFE is because of the missing authorized view instead of the missing table.
 
It's similar to when we are trying to write to a table and got an NotFound error, we wouldn't print the error msg as "Failed to write to non-existent table or instance"?

Reply:
Ok, make sense :)